lib/inspec/config.rb

```diff
@@ -0,0 +1,391 @@
+        transport_options.include? option_name # e.g., 'host'
+      end
+      credentials.merge!(unprefixed_transport_options)
+
+      # If there are any prefixed options, merge them in, stripping the prefix.
+      transport_prefix = transport_name.downcase.tr('-', '_') + '_'
+      transport_options.each do |bare_option_name|
+        prefixed_option_name = transport_prefix + bare_option_name.to_s
+        if final_options.key?(prefixed_option_name)
+          credentials[bare_option_name.to_s] = final_options[prefixed_option_name]
+        end
+      end
+    end
+
+    # fetch any info that applies to all transports (like sudo information)
+    def _utc_generic_credentials
+      @final_options.select { |option, _value| GENERIC_CREDENTIALS.include?(option) }
+    end
+
+    def _utc_determine_backend(credentials)
+      return if credentials.key?(:backend)
+
+      # Default to local
+      unless @final_options.key?(:target)
+        credentials[:backend] = 'local'
+        return
+      end
+
+      # Look into target
+      %r{^(?<transport_name>[a-z_\-0-9]+)://.*$} =~ final_options[:target]
+      unless transport_name
+        raise ArgumentError, "Could not recognize a backend from the target #{final_options[:target]} - use a URI format with the backend name as the URI schema.  Example: 'ssh://somehost.com' or 'transport://credset' or 'transport://' if credentials are provided outside of InSpec."
+      end
+      credentials[:backend] = transport_name.to_s # these are indeed stored in Train as Strings.
+    end
+
+    #-----------------------------------------------------------------------#
+    #                         Reading Config Files
+    #-----------------------------------------------------------------------#
+
+    # Regardless of our situation, end up with a readable IO object
+    def resolve_cfg_io(cli_opts, cfg_io)
+      raise(ArgumentError, 'Inspec::Config must use an IO to read from') if cfg_io && !cfg_io.respond_to?(:read)
+      cfg_io ||= check_for_piped_config(cli_opts)
+      return cfg_io if cfg_io
+
+      path = determine_cfg_path(cli_opts)
+
+      cfg_io = File.open(path) if path
+      cfg_io || StringIO.new('{ "version": "1.1" }')
+    end
+
+    def check_for_piped_config(cli_opts)
+      cli_opt = cli_opts[:config] || cli_opts[:json_config]
+      Inspec.deprecate(:cli_option_json_config, '') if cli_opts.key?(:json_config)
+
+      return nil unless cli_opt
+      return nil unless cli_opt == '-'
+      # This warning is here so that if a user invokes inspec with --config=-,
+      # they will have an explanation for why it appears to hang.
+      Inspec::Log.warn 'Reading JSON config from standard input' if STDIN.tty?
+      STDIN
+    end
+
+    def determine_cfg_path(cli_opts)
+      path = cli_opts[:config] || cli_opts[:json_config]
+      Inspec.deprecate(:cli_option_json_config, '') if cli_opts.key?(:json_config)
```

Comment:
Same as other feedback, is there a place to put this to so we don't have to repeat it?

Reply:
My intent is indeed to repeat it. `Inspec.deprecate` takes over the job of `# TODO: deprecate ...` comments, which we would have had no concerns with repeating each place the feature exists. When the time comes to remove that feature, we don't want to do a bunch of tracing to determine all changes that must be made; instead, rely on the easy grep-ability of the deprecation group name, `cli_option_json_config`.